src/composed/message/decrypt.rs

```diff
@@ -3,26 +3,73 @@ use log::debug;
 use crate::crypto::sym::SymmetricKeyAlgorithm;
 use crate::errors::Result;
 use crate::packet::SymKeyEncryptedSessionKey;
-use crate::types::{Mpi, SecretKeyRepr, SecretKeyTrait};
+use crate::types::{EskType, Mpi, SecretKeyRepr, SecretKeyTrait};
 
 /// Decrypts session key using secret key.
-pub fn decrypt_session_key<F, L>(locked_key: &L, key_pw: F, mpis: &[Mpi]) -> Result<PlainSessionKey>
+pub fn decrypt_session_key<F, L>(
+    locked_key: &L,
+    key_pw: F,
+    values: &EskBytes,
+    typ: EskType,
+) -> Result<PlainSessionKey>
 where
     F: FnOnce() -> String,
     L: SecretKeyTrait<Unlocked = SecretKeyRepr>,
 {
     debug!("decrypt session key");
 
-    locked_key.unlock(key_pw, |priv_key| {
-        let (key, sym_alg) = priv_key.decrypt(mpis, locked_key.fingerprint().as_bytes())?;
-        // TODO: handle other versions
-        Ok(PlainSessionKey::V4 { key, sym_alg })
-    })
+    locked_key.unlock(key_pw, |priv_key| priv_key.decrypt(values, typ, locked_key))
+}
+
+/// Values comprising the encrypted session key
+///
+/// FIXME: extend for algorithm specific values? (and/or v3 vs. v6)
```

Comment:
this sounds like a good plan, given they are differently structured depending on the version

Reply:
(Mostly independently, I just renamed `EskBytes` to `PkeskBytes` and moved it to `crate::types` in 6b66d8827f1913e253ff7d7618f0c28e6d6e7d14.
Maybe it should live in `crate::packet::public_key_encrypted_session_key`, though?)

That aside, I've thought about how to improve the shape of the enum a number of times now, and can't come up with a significant improvement.

I think the current enum models the algorithm-specific data reasonably.

The main oddity is that `SymmetricKeyAlgorithm` is an optional field for the `X25519` and `X448` variants. That field is only present for v3 PKESK, not for v6 PKESK.
Maybe the field name could/should be improved to make it clearer that this optionality is version-specific (currently only the docs explain this, not the field names).

Unless some epiphany strikes me unexpectedly, for now I'll give up trying to improve the enum, but would be very happy if you have ideas.

Reply:
seems fine for now